### DISCUSSION TITLE
Questions for *accepted* speakers

### DISCUSSION THREAD

**Discussion by Rafiot** [pretalx/pretalx#1066] [Ideas]
It would be nice to have a way to set questions specifically for the accepted speakers. In our case, that would be the banking information so we can reimburse them. 

I can do it by enabling questions after the review process and send a reminder to the accepted speakers, but it would be nice to have a specific type of questions.

**Comment by rixx** [comment]
We could add a question type for speakers as opposed to the current one that is for all submitters, but the only advantage would be that you could configure the questions in the beginning while setting up your event.

**Comment by Rafiot** [comment]
The other advantage is that it would be less error-prone (i.e. I'd be certain the only people receiving the mail are the ones with accepted presentations).

But yes, it's not a priority.

**Comment by rixx** [comment]
This is an issue that is well-suited for external contributors – let me know if I can be of any help, either via this issue or via support@pretalx.com.

## What this is about

pretalx provides a [Question](https://docs.pretalx.org/en/latest/developer/interfaces/models.html#questions-and-answers) model that allows to get arbitrary information, either per speaker, per submission, or per reviewer on reviews. It would be nice to have the option of asking only accepted speakers, additionally.

## How to get started

First of all, leave a comment on this issue to make sure nobody else starts working on this issue at the same time!

Next, you need pretalx running locally on your machine. First, [fork](https://github.com/pretalx/pretalx/fork) the repository to your own GitHub account. Then clone it locally (or [set it as your remote](https://help.github.com/articles/configuring-a-remote-for-a-fork/) if you already cloned it earlier)

Follow the [developer setup guide](https://docs.pretalx.org/en/latest/developer/setup.html), and then play around with pretalx a bit.
Have a look at the places impacting this issue. Look at the code responsible for the issues (searching for relevant words or files should help you find your way around). We have an explanation of our [code structure](https://docs.pretalx.org/en/latest/developer/concepts.html) and the actual [data models](https://docs.pretalx.org/en/latest/developer/interfaces/models.html).


## How to do this

The question model is defined in `pretalx/submission/models/questions.py`. You'll have to consider if an additional question target or a flag ("only accepted") would fit this use case better. The flag is probably the cleaner and most of all easier version to implement. Then you'll have to modify code in `QuestionForm` to take an `only_accepted` flag, and pass this argument whereever `QuestionForm` is used.

## Submit the PR

There are a number of tutorials on the Pull Request workflow online, for example this [short version](https://guides.github.com/introduction/flow/) by GitHub itself and this [more in-depth tutorial](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github).

Remember to provide documentation and tests, if applicable. Run the tests before opening your PR, including style checks. We have a [contribution guide](https://docs.pretalx.org/en/latest/developer/contributing.html), which ought to prove helpful at this stage.

**Comment by plaindocs** [comment]
Yeah, I'd love ti see this. 

It would also improve the experience when copying an event, because the questions would be more correct than copying the added questions and having to remove them before sending out the CFP
